src/openms/source/SYSTEM/UpdateCheck.cpp

```diff
@@ -0,0 +1,162 @@
+#include <QtCore/QDateTime>
+
+#include <OpenMS/DATASTRUCTURES/String.h>
+
+#include <OpenMS/CONCEPT/VersionInfo.h>
+
+using namespace std;
+  
+namespace OpenMS
+{
+  void UpdateCheck::run(const String& tool_name, const String& version)
+  {
+    String architecture = QSysInfo::WordSize == 32 ? "32" : "64";
+
+    // if the revision info is meaningful, show it as well
+    String revision("UNKNOWN");
+    if (!VersionInfo::getRevision().empty() && VersionInfo::getRevision() != "exported")
+    {
+      revision = VersionInfo::getRevision();
+    }
+    String platform;
+
+#ifdef OPENMS_WINDOWSPLATFORM
+    platform = "Win";
+#elif __APPLE__
+    platform = "Mac";
+#else
+    platform = "Linux";
+#endif
+
+    // write to tmp + userid folder
+
+    // e.g.: OpenMS_Default_Win_64_FeatureFinderCentroided_2.0.0
+    String tool_version_string;
+    tool_version_string = String("OpenMS") + "_" + "Default_" + platform + "_" + architecture + "_" + tool_name + "_" + version;
+
+    String version_file_name = File::getOpenMSHomePath() + "/.OpenMS/" + tool_name + ".ver";
+
+    // create version file if it doesn't exist yet
+    bool first_run(false);
+    if (!File::exists(version_file_name) || !File::readable(version_file_name))
+    {
+      Param p = File::getSystemParameters(); // initializes .OpenMS folder
+
+      // touch file to create it and set initial modification time stamp
+      QFile f;
+      f.setFileName(version_file_name.toQString());
+      f.open(QIODevice::WriteOnly);
+      f.close();
+      first_run = true;
+    }
+
+    if (File::readable(version_file_name))
+    {
+      QDateTime last_modified_dt = QFileInfo(version_file_name.toQString()).lastModified();
+      QDateTime current_dt = QDateTime::currentDateTime();
+
+      // check if at least one day passed sincle last request
+      if (first_run || current_dt > last_modified_dt.addDays(1))
+      {
+        // update modification time stamp
+        struct stat old_stat;
+        time_t mtime;
+        struct utimbuf new_times;
+        stat(version_file_name.c_str(), &old_stat);
+        mtime = old_stat.st_mtime; 
+        new_times.actime = old_stat.st_atime; // keep accessuib time unchanged 
+        new_times.modtime = time(NULL);  // mod time to current time
+        utime(version_file_name.c_str(), &new_times);          
+
+        LOG_INFO << "The OpenMS team is collecting use statistics for quality control and funding purposes." << endl;
+        LOG_INFO << "We will never give out your personal data but you may disable this functionality by setting the environmental variable OPENMS_DISABLE_USAGE_STATISTICS." << endl;
+        
+        // We need to use a QCoreApplication to fire up the  QEventLoop to process the signals and slots.
+        char* argv2[] = { "dummyname", NULL };
+        int argc = 1;
+        QCoreApplication event_loop(argc, const_cast<char**>(argv2));
+        NetworkGetRequest* query = new NetworkGetRequest(&event_loop);
```

Comment:
The examples I found always used new (seems to be a Qt thing. Actually, I think the QCoreApplication takes ownership of child objects and frees them at the appropriate time but I am not totally sure).  In this case I think it should work as stack object, too.